src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -468,17 +484,45 @@ private BoundExpression MakeConversionNode(BoundExpression rewrittenOperand, Typ
                      rewrittenOperand.Type.SpecialType != SpecialType.System_DateTime)
                 {
                     // error CS0029: Cannot implicitly convert type '{0}' to '{1}'
-                    _diagnostics.Add(
+                    diagnostics.Add(
                         ErrorCode.ERR_NoImplicitConv,
                         rewrittenOperand.Syntax.Location,
                         rewrittenOperand.Type,
                         rewrittenType);
                 }
+            }
 
-                return _factory.NullOrDefault(rewrittenType);
+            return conversion;
+        } 
+
+        private static BoundExpression MakeConversionNodeIfNecessaryWithoutLowering(
+            BoundExpression operand, 
+            TypeSymbol type, 
+            SyntaxNode syntax, 
+            CSharpCompilation compilation, 
+            DiagnosticBag diagnostics, 
+            bool @checked, 
+            bool acceptFailingConversion = false)
+        {
+            if (operand.Type.Equals(type))
+            {
+                return operand;
             }
+            Conversion conversion = MakeConversion(operand, type, compilation, diagnostics, acceptFailingConversion); 
 
-            return MakeConversionNode(rewrittenOperand.Syntax, rewrittenOperand, conversion, rewrittenType, @checked);
+            if (!conversion.IsValid)
+            {
+                return SyntheticBoundNodeFactory.NullOrDefault(type, syntax);
```

Comment:
>return SyntheticBoundNodeFactory.NullOrDefault(type, syntax); [](start = 16, length = 61)

This doesn't feel right, we are dropping the operand. Please add a test to cover scenario like that. I think we should always create BoundConversion, but mark it as "bad" if conversion doesn't exist.